### DISCUSSION TITLE
Multi user like Duo

### DISCUSSION THREAD

**Discussion by marcelmah** [ClassicOldSong/Apollo#94] [Ideas]
Hi,

I've been wanting to set up a local game hosting solution since I canceled my [Shadow PC (boost)](https://shadow.tech/shadowpc/offers#Configuration) over a year ago... got a fast laptop instead, but now I have no way of gaming on my 'TV' (NVIDIA Shield Pro 2019).

I had [Parsec ](https://parsec.app/) on my to do list (hoping it would be the easiest / most mature way) and last night (literally) I finally made/got the time.
It worked but the Android TV app sucks, I had to go through a lot of hoops to even login.
When I had it working I had it setup to connect to a local PC account "Family", which needs to be logged in.
So I lost local use of my laptop, but I was hoping I could still RDP to it (maybe even with [RDPwrap](https://github.com/sebaxakerhtc/rdpwrap) or something). Unfortunately Parsec logs off other users when you connect.

So I went looking for [Moonlight ](https://moonlight-stream.org/) and found [Sunshine](https://github.com/LizardByte/Sunshine). Works fine, but soon found [Duo ](https://github.com/DuoStream/Duo) being mentioned on Reddit somewhere.

Installed it and wow I was able to have Sunshine work under the Family user (and connect) while my normal user was untouched (even the console!).
But after two hours it stopped working and now I am unable to connect anymore, it tries, but after a while I get the firewall / open ports message.

Then I found Apollo... it works great and of all projects you look like the most active (recent). I've set it up and it seems to be working fine, but It runs under the logged in user.

So long story short... keep up the good work! and would you be interested in adding the features Duo has (did not use Sandbox on Duo), maybe even teaming up with that dev?

**Comment by ClassicOldSong** [comment]
Thanks for your appreciation but the answer is no and never.

Duo achieved multi user by reverse engineering and using RDPWrap, which is 

1. Not sustainable 
2. Have potential security issues 
3. Some apps have already reported not opening in RDP context 
4. Potentially in violation of Windows TOS

Apollo uses legit Windows API for virtual display creation, and to applications the virtual display isn't virtual at all. The goal isn't multi user but multi display with Apollo.

The two projects have way different goals, and Apollo will never look into multi user usage unless Windows provides a legitimate way to do this.

**Reply by ClassicOldSong** [reply]
Duo is also in potential violation of the GPL license, and it's paid and close sourced while Apollo remains free and open source. There's no chance of cooperation unless Duo open source its code base.

**Reply by marcelmah** [reply]
Hi,

Thank you for your quick reply.

I understand, I will use this information in choosing the solution I will use (long term).
If I can get Apollo working nicely (and the wife and kids can use it) I will go this route.

Looking forward to what you make of it!
